### ISSUE TITLE
swupd - command line enhancements

### ISSUE THREAD

**Issue by rbharathkumar** [clearlinux/swupd-client#805]
The command line for installing and updating is quite useful, but it can be improved.

**Describe the solution you'd like**
Please consider the following enhancements.
1. On firing off the `sudo swupd update` command, the update process starts immediately and does not wait for any further user confirmation. 
Can this be improved as follows ?

On firing the above said command, can we get a list of bundles to be updated, their current version say 27600, the version to which it is to be updated 27660, the download size of each bundle, the net change to size after installation ? then it can wait for user confirmation.

When the user presses y then the process can go ahead with the usual output that we have like bundles, files modified, files deleted, maybe the download progress can be made graphical like a moving dashes, just enhancements to keep the end-user aware of what is happening.

Also it is very useful in cases like mine where the data is metered and costly. I may want to hold off on updating till my billing cycle restarts again. So if I knew the download size it would be very useful.

**Comment by rbharathkumar** [comment]
Yes, I understand that not updating or partial updates are not advisable in any rolling release system.
As you rightly put it, it may cause instability or worse security issues.

My point was to defer it by a few hours or maybe the next day, purely around the metering issue.
In my case I wanted to update and test the new plasma updates, but had to wait for a couple of days to get it updated. Also just curious, do we have the option to update a specific bundle alone ?

**Comment by otaviobp** [comment]
Related to #647

**Comment by otaviobp** [comment]
> My point was to defer it by a few hours or maybe the next day, purely around the metering issue.

You are talking about the autoupdate, right? We had some suggestions on creating a snooze button for auto-updates, but we didn't move forward with that. Is this your proposal?

> In my case I wanted to update and test the new plasma updates, but had to wait for a couple of days to get it updated. Also just curious, do we have the option to update a specific bundle alone ?

No. You can force one specific bundle to move to one specific clear version using swupd verify --fix command with option -B, but it's dangerous and I don't recommend using that other than for tests purposes. And that command would also install all included bundles, so it may be more than you would like.

**Comment by rbharathkumar** [comment]
Thank you for your response @otaviobp 

Kind of yes and no..
An snooze on the auto-update would be nice. But that was not my only proposal.

My proposal was more on the usability front.

1. swupd should be able to provide a clean list of bundles to be updated, their current version, to be updated version **and the amount of data to be downloaded.** This information is really important for users who are on metered connections.

2. I do agree that swupd has a check-update option, but user usually go for the swupd update command directly. So would be nice if swupd prompted users if they want to go ahead or not after displaying a list as mentioned in point 1.

Again the point two is purely based on the data consumption part only and not meant as a means to avoid updates. As a rolling release system I understand the issues that are caused by not updating a system regularly and hence re-iterating that this proposal is merely on the visual output front and not a means to avoid updates.

3. As referenced in some of the other approach, would be nice to have a way of tracking the progress of the updates similar to how pacman or dnf do it.

4. The messages received at the end of the update requesting for a restart are a little confusing, do they mean the restart of the system or the service. (Not sure if this still persists)..

Kindly do let me know for any further information...
